waku/v2/discv5/discover.go

```diff
@@ -291,7 +287,7 @@ func (d *DiscoveryV5) iterate(ctx context.Context, iterator enode.Iterator, limi
 
 	for {
 		if len(d.peerCache.recs) >= limit {
-			break
+			time.Sleep(1 * time.Minute)
```

Comment:
I'm not very familiar with the code, thus question:
previously we would break the cycle and now we just delay execution, what's the point?

Reply:
Previously, if we reached the maximum number of discovered nodes (600), we would stop discovering new peers, and exit the goroutine.  Calling `FindPeers` would still work for 1hr since peers would still be in the cache, however, after 1 hour the peers would be removed from the cache, and no peers would be discovered since the iterate() function would never be executed

Reply:
@richard-ramos btw does this affect this issue https://github.com/status-im/status-go/issues/3063 ? Does it mean that there is no need to restart discovery as it is not stopped now?

Reply:
I think it does not. It would help once you discover 600 peers, since discovery would not stop. I have some code in progress that would probably help with 3063